src/dispatching/update_listeners/webhooks.rs

```diff
@@ -71,6 +86,32 @@ impl Options {
     pub fn drop_pending_updates(self) -> Self {
         Self { drop_pending_updates: true, ..self }
     }
+
+    /// A secret token to be sent in a header “X-Telegram-Bot-Api-Secret-Token”
+    /// in every webhook request, 1-256 characters. Only characters `A-Z`,
+    /// `a-z`, `0-9`, `_` and `-` are allowed. The header is useful to ensure
+    /// that the request comes from a webhook set by you.
+    ///
+    /// ## Panics
+    ///
+    /// If the token is invalid.
+    #[track_caller]
+    pub fn secret_token(self, token: String) -> Self {
+        check_secret(token.as_bytes()).expect("Invalid secret token");
+
+        Self { secret_token: Some(token), ..self }
+    }
+
+    /// Returns `self.secret_token`, generating a new one if it's `None`.
+    ///
+    /// After a call to this function `self.secret_token` is always `Some(_)`.
+    ///
+    /// **Note**: if you leave webhook setup to teloxide, it will automatically
+    /// generate a secret token. Call this function only if you need to know the
+    /// secret (for example because you are calling `set_webhook` by yourself).
+    pub fn get_or_gen_secret_token(&mut self) -> &str {
```

Comment:
Also `#[must_use]`?

Reply:
I don't think that this function should be `#[must_use]`, after all it has side effects.